### PR TITLE
Introduce onboarding wizard

### PR DESCRIPTION
**Pull request by shervElmi**
This PR introduces a new "Onboarding Wizard" feature within the AWP Classified Plugin Admin Panel.

#### First Method:

Activate the AWP Classified plugin from the WP Admin Plugins page. Please note that you will be redirected to the Onboarding Wizard page **only if this is your first time activating the plugin**. Otherwise, you can access it directly via the URL.

#### Second Method:

Directly access the Onboarding Wizard via the URL.
Example: `http://example.com/wp-admin/admin.php?page=awpcp-recipes-onboarding-wizard`

### Demo

CleanShot 2024-11-19 at 18 03 31

### CODE REVIEW

**Comment by shervElmi** [comment]
Hi @sorinmarta,  

There’s an issue with the plugin’s activation process that is beyond the scope of this PR.  

I’ve added the onboarding-related activation code to `installer.php` as shown below:  

```PHP
public function activate() {
    $this->install_or_upgrade();

    update_option( 'awpcp-activated', true );

    if ( get_transient( AWPCP_OnboardingWizard::TRANSIENT_NAME ) !== 'no' ) {
        set_transient(
            AWPCP_OnboardingWizard::TRANSIENT_NAME,
            AWPCP_OnboardingWizard::TRANSIENT_VALUE,
            60
        );
    }
}
```

However, the activation hook is currently disabled in the plugin. I attempted to enable it by uncommenting the code below, but it caused a PHP fatal error. I also tried resolving the issue, but it falls outside the scope of this PR.  

Here’s the relevant code from `includes/class-awpcp.php`:  

```PHP
public function bootstrap() {
    $this->settings_manager = $this->container['SettingsManager'];
    $this->js = AWPCP_JavaScript::instance();

    awpcp_load_plugin_textdomain( AWPCP_FILE );

    // TODO: Fix activation hook
    // awpcp_register_activation_hook( AWPCP_FILE, array( $this->installer, 'activate' ) );
}
```

**Comment by shervElmi** [comment]
Because of this issue, when activating the plugin, users are not redirected to the Onboarding Wizard page. We need to resolve this issue first so I can test my code and proceed with reviewing this PR.

**Comment by sorinmarta** [comment]
@shervElmi 

That activation logic is a big mess that will require a lot of time to fix so for efficiency I think we should just register an activation hook that does what you need so you can proceed.

I pushed a commit that does that, so you can test now.

Let me know when its tested and confirmed so that I can review.

Thanks!

**Comment by shervElmi** [comment]
Thanks for your work on the issue @sorinmarta. The PR is now ready for review.

File: functions.php [hunk 1]
```diff
@@ -3242,3 +3242,40 @@ function awpcp_get_server_ip_address_from_httpbin() {
...
+/**
+ * Outputs or retrieves sanitized inline SVG content.
+ *
+ * @since x.x
+ * @param string $path       Relative or absolute path to the SVG file.
+ * @param bool   $use_images Whether to prepend the images directory to the path. Default true.
+ * @param bool   $echo       Whether to echo the output or return it. Default true.
+ * @return string|void
+ */
+function awpcp_inline_svg( $path, $use_images = true, $echo = true ) {
```
Comment (sorinmarta) [inline functions.php:3255]: The doc comment says that the function sanitizes the SVG content but I don't see any sanitization.

File: includes/admin/class-admin.php [hunk 2]
```diff
@@ -162,6 +162,28 @@ public function is_awpcp_post_page() {
         return $post_type === $this->post_type;
     }
 
+    /**
+     * Check for certain page in Formidable settings
```
Comment (sorinmarta) [inline includes/admin/class-admin.php:166]: We don't check in Formidable settings 😅

File: resources/js/onboarding-wizard/ui/setupInitialView.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ */
+function navigateToInitialStep() {
+	const initialStepName = determineInitialStep();
+
+	clearOnboardingQueryParams();
+	navigateToStep( initialStepName, 'replaceState' );
+}
+
+/**
+ * Determines the initial step based on the current state, such as whether Formidable Pro is installed
```
Comment (sorinmarta) [inline resources/js/onboarding-wizard/ui/setupInitialView.js:34]: AWPCP pro, not Formidable Pro. And same for the function above

**Review by sorinmarta — CHANGES_REQUESTED** [review]
Great work as always @shervElmi! I added a few small comments and questions. Mostly about code comments, not functionality.

I'll leave the decision to change them or not to you. But before merging we need to fix the CI issues.

Thanks!

**Comment by shervElmi** [comment]
I'm happy with your positive feedback and truly appreciate it @sorinmarta. I have just addressed the issues with the comments.

**Comment by shervElmi** [comment]
@sorinmarta, could you check the PHPStan errors? They don't seem to be related to this PR. I haven't reviewed them in detail, so it's possible that some may be relevant to this PR.

**Review by sorinmarta — APPROVED** [review]
Thanks @shervElmi! You are right, there were some issues with the PHPStan config we have. It's all good now so I'll get this merged soon.

Amazing work!

**Pull request merged**